src/replication.c

```diff
@@ -1582,13 +1582,7 @@ void sendBulkToSlave(connection *conn) {
     }
 
     /* If the preamble was already transferred, send the RDB bulk data. */
-    if (lseek(slave->repldbfd,slave->repldboff,SEEK_SET) == -1) {
-	serverLog(LL_WARNING,"Failed to lseek the RDB file to offset %lld for replica %s: %s",
-	    (long long)slave->repldboff, replicationGetSlaveName(slave), strerror(errno));
-	freeClient(slave);
-	return;
-    }
-    buflen = read(slave->repldbfd,buf,PROTO_IOBUF_LEN);
+    buflen = pread(slave->repldbfd,buf,PROTO_IOBUF_LEN,slave->repldboff);
```

Comment:
pread doesn't change the offset of this file, but lseek does.
I prefer not to change this code unless it can obviously bring benefits.
@StavRLevi @moticless

Reply:
Agree.